rewrite-groovy/src/main/java/org/openrewrite/groovy/GroovyParserVisitor.java

```diff
@@ -2118,11 +2118,16 @@ private <T extends TypeTree & Expression> T typeTree(@Nullable ClassNode classNo
         return expr.withPrefix(prefix);
     }
 
-    private List<JRightPadded<Space>> arrayDimensionsFrom(ClassNode classNode) {
-        List<JRightPadded<Space>> result = new ArrayList<>();
+    private List<J.ArrayType.AnnotatedDimension> arrayDimensionsFrom(ClassNode classNode) {
+        List<J.ArrayType.AnnotatedDimension> result = new ArrayList<>();
         while(classNode != null && classNode.isArray()) {
             classNode = classNode.getComponentType();
-            result.add(JRightPadded.build(sourceBefore("[")).withAfter(sourceBefore("]")));
+
+            // TODO: add support for getting each dimensions' annotations from classNode
```

Comment:
Should we associate an issue with this TODO item and link it here such that it's not lost?
And what's the impact of not yet covering this in the current implementation?

Reply:
The impact of not covering this should just be that the same bug will appear in Groovy annotated arrays. I'm not as familiar with the Groovy language specification, and the `GroovyParserVisitor` is written differently to the `JavaParserVisitor`s, so I opted to not implement it for now. It should be very similar to the Java case though; the `AnnotatedDimension` structure is already there so it should be easy to implement.